client.go

```diff
@@ -248,20 +252,24 @@ func (r remoteProcess) Resize(ctx context.Context, rows, cols uint16) error {
 	return r.conn.Write(ctx, websocket.MessageBinary, payload)
 }
 
-func (r remoteProcess) Wait() error {
-	select {
-	case err := <-r.done:
-		return err
-	case <-r.ctx.Done():
-		return r.ctx.Err()
+func (r *remoteProcess) Wait() error {
+	<-r.done
+	if r.readErr != nil {
+		return r.readErr
+	}
+	// when listen() closes r.done, either there must be a read error
+	// or exitCode is set non-nil, so it's safe to dereference the pointer
+	// here
+	if *r.exitCode != 0 {
+		return ExitError{Code: *r.exitCode}
 	}
```

Comment:
Although true, it can never hurt to be defensive right?

Reply:
I'm actually not a fan of defensiveness in this context---that is, bugs in our own code.  If we do something defensive, and drop a log or return an error, it's super easy for that bug to be ignored.  It's much harder to ignore a segfault.

The defensiveness we should be practicing is extensive unit testing, so that we will catch any logic errors before they get checked into master.